docs/docs/fault-domain-awareness.md

```diff
@@ -0,0 +1,100 @@
+	 
+You can deploy your apps or pods in a specific region region based on the available capacity.
+
+### Local and remote regions
+
+- A **local region** is a region that contains the master nodes.
+- A **remote region** contains only agent nodes. There is high latency between a remote region and the local region.
+
+## Mesos installation considerations
+
+- Masters nodes must be in the same region because otherwise the latency between them will be too high. They should, however, be spread across zones for fault-tolerance.
+	
+- You must have less than 100 ms latency between regions.
+
+## Adding fault domain awareness to your app or pod definition
+
+In your Marathon app or pod definition, you can use placement constraints to:
+
+- Specify a region and zone for your app or pod, so that all instances of the app or pod will be scheduled only in that region and zone.
+
+- Specify a region without a specific zone, so that all instances of a given app or pod will be scheduled in that region (but not necessarily in the same zone).
+
+### Placement constraint guidelines
+
+- If no region is specified in your app or pod definition, instances are only scheduled for the local region because of high latency between the local and remote region. No instance will ever be scheduled for an agent outside of the local region unless you explicitly specify.
+
+- If a region is specified without a specific zone, instances are scheduled on any agent in the given region.
+
+- If both region and zone are specified, instances are scheduled on any agent in the given region and zone, and not in any other region or zone.
+
+- If you specify a hostname `UNIQUE` constraint, that constraint is also honored in remote regions.
+
+### Examples
+
+Suppose we have a Mesos cluster that spans 3 regions: `aws-us-east1`, `aws-us-east2`, and `local`. Each region has zones `a`,`b`,`c`,`d`.
+
+#### Specify only a remote region
+
+```
+{
+  "instances": 5,
+  "constraints": [
+    ["@region", "IS", "aws-us-east1"]
+  ]
+}
+```
+
+- No instance will launch in the local region.
+- All of the 5 instances will be launched in the `aws-us-east1` region.
+
+#### Balanced Placement for a Single Region
+
+```
+{
+   ...
+  "instances": 6,
+  "constraints": [
+    ["@region", "IS", "aws-us-east1"],
+    ["@zone", "GROUP_BY", "4"]
+  ]
+}
+```
+
+- Instances will all be launched in the `aws-us-east1` region and evenly divided between `aws-us-east1`‘s zones `a`,`b`,`c`,`d`.
+
+### Use Case: Use Remote Regions to Increase Capacity
+
+You can configure apps and pods to use remote regions when extra capacity is needed. Your cluster must consist of one local region, which contains the master agents, system services, and agents, and one or more remote regions, which will contain only agents.
+
+To increase capacity, you will add new agents to a remote region or regions of your cluster, and then update your apps or pods to be launched in that region or those regions appropriately.
```

Comment:
I might suggest that we take out any reference to launch in multiple regions. Sadly, due to time constraints and needing to sync with other products supporting the constraint language, we did not implement the IN constraint operator. Launching an app in multiple regions, right now, requires that the operator write a regular expression. I don't know if we want to even mention that, or provide a guide to do so. Regular require a high degree of expertise to get correct.

Reply:
I would prefer that we DO mention it... that it is an advanced feature that requires regex at the moment.   As a user, I would rather have transparency from the product team and understand what is possible.   Perhaps it will drive the "IN" feature request externally as well.

Reply:
In case I do end up mentioning this functionality, can either of you tell me more about what the regex needs to do?

Reply:
Latest commit addresses all comments except this one -- I'm checking with @vinodkone later to see how to proceed.